### PR TITLE
Added equal, unique and deepcopy functions

### PR DESCRIPTION
**Pull request by Sanyuan-Chen**
Tried to implement these three functions for namedtensor, hope it can be helpful! :)

### CODE REVIEW

File: namedtensor/test_core.py [hunk 1]
```diff
@@ -33,6 +33,45 @@ def test_stack_basic(data, x):
...
+    import copy
+    x2 = copy.deepcopy(x)
+    assert id(x2.values) != id(x.values)
+    assert torch.equal(x2.values, x.values)
+    assert id(x2._schema) != id(x._schema)
+    assert x2._schema._names == x._schema._names
+    assert x2._schema._masked == x._schema._masked
+
+
+@given(data(), named_tensor())
```
Comment (srush) [inline namedtensor/test_core.py:48]: Yay! Way to use the new testing.

File: namedtensor/test_core.py [hunk 2]
```diff
@@ -33,6 +33,45 @@ def test_stack_basic(data, x):
...
+def test_unique(data, x):
+    s = data.draw(dim(x))
+    nu = data.draw(name(x))
```
Comment (srush) [inline namedtensor/test_core.py]: There should be a `names` function where you can draw 2 (max_size=2) that will guarantee they are unique.

File: namedtensor/torch_base.py [hunk 3]
```diff
@@ -99,6 +99,43 @@ def cat(tensors, dim):
                 tensors[i] = tensors[i]._force_order(tensors[0]._schema._names)
         return tensors[0]._new(torch.cat([t.values for t in tensors], dim=dim))
 
+    @staticmethod
+    def equal(tensor1, tensor2):
```
Comment (srush) [inline namedtensor/torch_base.py:106]: I think we need to be more careful with the semantics here. The two tensors need to first have the same order before we check equality.

File: namedtensor/torch_helpers.py [hunk 4]
```diff
@@ -342,7 +342,6 @@ def __dir__(self):
         "expm1",
         "float",
         "floor",
-        "fmod",
```
Comment (srush) [inline namedtensor/torch_helpers.py]: Be sure to rebase from master so you get this line back.

File: namedtensor/torch_base.py [hunk 5]
```diff
@@ -99,6 +99,43 @@ def cat(tensors, dim):
...
+            return False
+
+    @staticmethod
+    def unique(
+        input,
+        sorted=False,
+        return_inverse=False,
+        dim=None,
+        names=("unique", "Indices"),
+    ):
```
Comment (srush) [inline namedtensor/torch_base.py]: Can you comment this function. It looks a bit complicated.

File: namedtensor/torch_base.py [hunk 6]
```diff
@@ -99,6 +99,43 @@ def cat(tensors, dim):
...
+        ):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def unique(
+        input,
+        sorted=False,
+        return_inverse=False,
```
Comment (srush) [inline namedtensor/torch_base.py]: Ideally these are just **kwargs unless the function needs them.

**Review by srush — CHANGES_REQUESTED** [review]

File: namedtensor/torch_helpers.py (hunk 4 above)
Comment (srush) [reply namedtensor/torch_helpers.py]: Oops. sorry this is fine.

File: namedtensor/torch_base.py (hunk 3 above)
Comment (Sanyuan-Chen) [reply namedtensor/torch_base.py:106]: Yeah, that makes much more sense. Do you think we should consider the 'masked' equality in this function, as the tensors which have the same names and 'masked' value but different names order would mask different dimensions?

File: namedtensor/torch_helpers.py (hunk 4 above)
Comment (Sanyuan-Chen) [reply namedtensor/torch_helpers.py]: Well, I deleted this line because I think 'fmod' is already in the self._noshift_args list, and putting it here caused the test_noshift fail.

**Review by srush — APPROVED** [review]

**Comment by srush** [comment]
Let me know if there are any outstanding questions, otherwise would be happy to merge.

**Comment by Sanyuan-Chen** [comment]
No questions. 👌
